### PR TITLE
Bump scala-cli to 1.4.1 + drop old cli management

### PR DESCRIPTION
**Pull request by hamzaremmal**
cc @WojciechMazur @Kordyjan

### CODE REVIEW

**Review by Gedochao — APPROVED** [review]
LGTM

File: dist/bin-native-overrides/cli-common-platform
Comment (WojciechMazur) [inline dist/bin-native-overrides/cli-common-platform:3]: I wonder do we really need this `cli-common-platform` files anymore. There seems to be exactly 1 usage for each of them

**Review by WojciechMazur — APPROVED** [review]
Looks good

File: dist/bin-native-overrides/cli-common-platform
Comment (hamzaremmal) [reply dist/bin-native-overrides/cli-common-platform:3]: We need to keep these 2 files separated because we have different content depending on what we are building. If we build universal archives (with jars and no native binaries), we will be using https://github.com/scala/scala3/blob/d48ba033d8128e04727dbee88a5a8eb3fa98d26d/dist/bin/cli-common-platform and https://github.com/scala/scala3/blob/main/dist/bin/cli-common-platform.bat. While if we are building archives with native binaries, we will be using https://github.com/scala/scala3/d48ba033d8128e04727dbee88a5a8eb3fa98d26d/main/dist/bin-native-overrides/cli-common-platform and https://github.com/scala/scala3/blob/d48ba033d8128e04727dbee88a5a8eb3fa98d26d/dist/bin-native-overrides/cli-common-platform.bat

**Comment by hamzaremmal** [comment]
I will wait for @bishabosha's review before merging this. unless we have some time constraints for backporting (@WojciechMazur ?)

**Comment by WojciechMazur** [comment]
> I will wait for @bishabosha's review before merging this. unless we have some time constraints for backporting (@WojciechMazur ?)

There won't be a 3.5.1-RC2 before 3.5.0 is published, so don't worry about the backports

**Review by bishabosha — APPROVED** [review]

**Pull request merged**
